src/app/core/services/ledger.service.ts

```diff
@@ -1,30 +1,35 @@
 import { Injectable } from '@angular/core';
-import { runTransaction } from '@numbersprotocol/niota';
-import { from, Observable, of } from 'rxjs';
-import { catchError, tap, timeout } from 'rxjs/operators';
+import { sendMessage } from '@numbersprotocol/niota';
+import { Observable, of } from 'rxjs';
+import { catchError, map, switchMap, take, tap, timeout } from 'rxjs/operators';
+import { WebCryptoApiSignatureProvider } from '../../core/services/web-crypto-api-signature-rovider.service'
 
 @Injectable({
   providedIn: 'root',
 })
 export class LedgerService {
-  constructor() {}
+  constructor(
+    private readonly CryptoSignature: WebCryptoApiSignatureProvider
+  ) {
+    this.CryptoSignature.initialize();
+  }
 
   private registerOnLedger(hash: string): Observable<any> {
-    const address =
-      'HEQLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWOR99D';
-    const seed =
-      'PUEOTSEITFEVEWCWBTSIZM9NKRGJEIMXTULBACGFRQK9IMGICLBKW9TTEVSDQMGWKBXPVCBMMCXWMNPDX';
+    const index = 'Lifebox';
     const rawMsg = { hash };
-    return from(runTransaction(address, seed, rawMsg)).pipe(
+    return this.CryptoSignature.getCryptoData().pipe(
+      switchMap((dummySignature) => sendMessage(index, rawMsg, dummySignature.publicKey)),
       timeout(10000),
-      tap(resultHash =>
-        console.log(`Hash ${resultHash} registered on ledger`, resultHash)
+      tap(resultHash => {
+        console.log(`Message ID ${resultHash} registered on ledger`)
```

Comment:
```suggestion
      tap(messageId => {
        console.log(`Message ID ${messageId} registered on ledger`)
```

Naming should change since the returned value is better described as a message ID instead of a result hash in IOTA 1.5